cmd/kubeadm/app/preflight/checks.go

```diff
@@ -415,14 +416,15 @@ func (hst HTTPProxyCheck) Name() string {
 // Check validates http connectivity type, direct or via proxy.
 func (hst HTTPProxyCheck) Check() (warnings, errors []error) {
 
-	url := fmt.Sprintf("%s://%s:%d", hst.Proto, hst.Host, hst.Port)
+	hostPort := net.JoinHostPort(hst.Host, strconv.Itoa(hst.Port))
+	url := fmt.Sprintf("%s://%s", hst.Proto, hostPort)
```

Comment:
This still builds URLs with strings, please use `(&url.URL{Scheme: hst.Proto, Host: hostPort}).String()` instead

Reply:
ok, will do